### PR TITLE
Move build system to Maven (retains Ant tasks for backwards compatibility)

### PR DESCRIPTION
**Pull request by michaelklishin**
Originally submitted by @PatrickSauts as #119.

This moves build system to Maven while retaining backwards compatibility by offering "proxy" Ant tasks.

### CODE REVIEW

**Comment by dumbbell** [comment]
Fixes #37.

**Comment by dumbbell** [comment]
The SSL testsuite does bot work because `isSSlAvailable()` crashes for me and, after when this is fixed, it returns false because `sslClientPath` points to a directory, not a file. I'm working on it.

**Comment by PatrickSauts** [comment]
Super merci !

**Comment by dumbbell** [comment]
A bit more details about my last commit: what was done in `AbstractRMQTestSuite.java` was in fact never called. Maybe because we still rely on JUnit 3 API, not 4. For example, the `runTest()` was never called so the properties initialized and verified in `AbstractRMQTestSuite.java` was ineffective.

This probably remained unnoticed because default values for `rabbitmqctl.bin` and `sibling.rabbitmq_test.dir` are ok when the Java client is inside the Umbrella. However, the default value for `make.bin` only works when GNU Make is installed as `make`, which is not the case on *BSD (GNU Make is available as `gmake`).

Our testsuite didn't handle well failures from `make`. Usually, this means the testsuite is stuck. I extended what @PatrickSauts did and added calls from the TestSuites themselves, to be sure we have all properties set before running tests.

Finally, the new `broker.{hostname,port,sslport}` properties were good when using Maven, but not Ant. That's what confused me yesterday. Now, the top-level `build.xml` explicitely sets those properties and `isSSLAvailable()` is back on rails.

**Comment by dumbbell** [comment]
Hmm, there is one more thing to fix: `ClientVersion.java` is correctly generated by `pom.xml` using the `impl.version` property (set in the top-level `Makefile`), but `pom.xml` hard-codes the version for itself (currently set at `3.6.1-SNAPSHOT`).

**Comment by dumbbell** [comment]
@PatrickSauts: Merged, thank you very much!

**Comment by PatrickSauts** [comment]
@dumbbell 
When moving completely from ant 
Uncomment line 148 and remove line 147 in pom.xml


Have the right `settings.xml` in your `~/.m2` directory like
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
<servers>
    <server>
      <id>sonatype-nexus-staging</id>
      <username>my_login</username>
      <password>my_password</password>
    </server>
  </servers>
</settings>
```

To release a new version the maven commands would be:
`mvn versions:set -DnewVersion=<new.version> && mvn versions:commit && mvn clean deploy`

It would set the version in the pom and use that version in the generated java file.


By the way you can also run the perf tests like this
`mvn -q exec:java -Dexec.classpathScope=test -Dexec.mainClass="com.rabbitmq.examples.PerfTest"`

or with arguments
```
mvn -q exec:java -Dexec.classpathScope=test -Dexec.mainClass="com.rabbitmq.examples.PerfTest" -Dexec.args="arg0 arg1 arg2" 
```

Basically any class with a main inside  package com.rabbitmq.examples without the need to moving any jar around, the classpath is defined by maven.

I added a `runperftestMaven.sh` and `stresspersisterMaven.sh` inside the script folder.

One last thing I can write an assembly to build a zip to run perf tests that would contain all the necessary files.

**Comment by michaelklishin** [comment]
@PatrickSauts note that we also distribute the JAR directly from GitHub and rabbitmq.com. But

```
mvn versions:set -DnewVersion=<new.version> && mvn versions:commit
```

certainly sounds right up our alley, thank you!

**Comment by PatrickSauts** [comment]
@michaelklishin You probably want to build the jars using maven too right ?

`mvn versions:set -DnewVersion=<new.version> && mvn versions:commit && mvn clean package`

I'm pretty sure that was implicit in your comment but just in case.

So moving ~completely away from ant you could get rid of `lib` and 'bundlor` folders, all bundlor files and build.xml.

Your very welcome. Glad I could help.

**Comment by michaelklishin** [comment]
@PatrickSauts sure, thank you. We will ping you once we have more Maven questions (which should be around the time `3.7.0` Milestone 2 ships, so in about a month).
